### PR TITLE
Mh/bugfix/get full app path

### LINKED ISSUE #130: [BUG] ConductorConfig.class:getFullAppPath() throws error if appFile ends in .app
## The problem
The new logic in the `getFullAppPath()` method checks the `appFile` specified in the yaml, and checks if the appFile is a directory. Files that end in `.app` are being interpreted as directories and causing an error when we try to list the files in the directory.

## Environment

* Conductor-Mobile Version: 0.19.0
* Last Conductor-Mobile version that did not exhibit the issue (if applicable): 0.18.0
* Desktop OS/version used to run Conductor-Mobile:
* Mobile platform/version under test: iOS 11
* Real device or emulator/simulator: simulator
* `appium-doctor` results are passing []: yes

## Details
I'm not sure why `.app` files are returning true for `isDirectory()`, but we should figure out why that is and either use a different method, or add an extra condition for seeing if the app/directory file ends in `.app`

## Link to Appium/Conductor-Mobile logs

Create a [GIST](https://gist.github.com) which is a paste of your _full_ Stacktrace, Appium logs, and `config.yaml` contents. Link them here.
Do _NOT_ paste your full logs here, as it will make this issue very long and hard to read!
If you are reporting a bug, _always_ include Relevant logs, please!

## Code To Reproduce Issue [ Good To Have ]
```
public String getFullAppPath() {
        if (this.appFile == null) {
            return null;
        } else {
            File dir = new File(this.appFile);
            Path path = Paths.get(this.appFile);
            if (dir.isDirectory()) {
                try {
                    String expectedFileExtension = this.platformName == Platform.ANDROID ? ".apk" : ".app";
                    File[] files = dir.listFiles((dir1, name) -> {
                        return name.toLowerCase().endsWith(expectedFileExtension);
                    });
                    if (files != null && files.length == 1) {
                        File fileFoundMatchingExtension = files[0];
                        return Paths.get(System.getProperty("user.dir"), fileFoundMatchingExtension.toString()).normalize().toString();
                    }

                    throw new IllegalArgumentException();
                } catch (IllegalArgumentException var7) {
                    Logger.error(var7, "Unable to find a valid application in specified appFile directory: {} . Make sure there is only ONE unique file extension in {}, or else you must specify an absolute path. Also, double check that there is indeed an application with a valid extension in the specified appFile directory. ERROR", new Object[]{dir, dir});
                }
            } else if (this.appFile.startsWith("sauce-storage:") || path.isAbsolute()) {
                return this.appFile;
            }

            try {
                new URL(this.appFile);
                return this.appFile;
            } catch (MalformedURLException var6) {
                return Paths.get(System.getProperty("user.dir"), path.normalize().toString()).normalize().toString();
            }
        }
    }
```

`appFile: ./apps/IOS/IOS.app`

### PR DESCRIPTION
**Pull request by mrk-han**
### Related Issue

closes #130 

### Description

Along with `.app` exception, this also fixes sauce storage exceptions that were being thrown if using sauce storage.

Also improved the error thrown when finding app given a directory

### Checklist

* [X] Have you added an explanation of what your changes do?
* [X] Have you written new tests for your core changes, as applicable?
* [X] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: src/main/java/com/joss/conductor/mobile/ConductorConfig.java [hunk 1]
```diff
@@ -540,14 +541,35 @@ public Platform getPlatformName() {
     }
 
     public String getFullAppPath() {
+
+        File dir = new File(appFile);
+        Path path = Paths.get(appFile);
```
Comment (aaron-goff) [inline src/main/java/com/joss/conductor/mobile/ConductorConfig.java]: Will we run into any exceptions if we don't do the null check before trying to set the `path`?

File: src/main/java/com/joss/conductor/mobile/ConductorConfig.java [hunk 2]
```diff
@@ -540,14 +541,35 @@ public Platform getPlatformName() {
...
+
+        //if appFile end with these, assume we should try to use them
+        if ((appFile.endsWith(".ipa") || appFile.endsWith(".apk") || appFile.endsWith(".app"))) {
+            return Paths.get(System.getProperty("user.dir"), path.normalize().toString()).normalize().toString();
+        }
+
         if (dir.isDirectory()) {
             try {
-                String expectedFileExtension = platformName == Platform.ANDROID ? ".apk" : ".app";
+                String expectedFileExtension = platformName == Platform.ANDROID ? ".apk" : ".app"; // =
```
Comment (aaron-goff) [inline src/main/java/com/joss/conductor/mobile/ConductorConfig.java]: Missing comment?

Comment (mrk-han) [reply src/main/java/com/joss/conductor/mobile/ConductorConfig.java]: Nice -- fixed in a9574c3

File: src/main/java/com/joss/conductor/mobile/ConductorConfig.java (hunk 1 above)
Comment (mrk-han) [reply src/main/java/com/joss/conductor/mobile/ConductorConfig.java]: Good call. e428235

**Pull request merged**